include/ensmallen_bits/adam/adamw_update.hpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file adamw_update.hpp
+ * @author Niteya Shah
+ *
+ * AdamW optimizer. Adam is an an algorithm for first-order gradient-based
+ * optimization of stochastic objective functions, based on adaptive estimates
+ * of lower-order moments.
+ *
+ * AdamW is extension of Adam that decouples the weight Decay segment.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_ADAM_ADAMW_UPDATE_HPP
+#define ENSMALLEN_ADAM_ADAMW_UPDATE_HPP
+
+#include "adam_update.hpp"
+
+namespace ens {
+
+/**
+ * Adam is an optimizer that computes individual adaptive learning rates for
+ * different parameters from estimates of first and second moments of the
+ * gradients as given in the section 7 of the following paper.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{Kingma2014,
+ *   author  = {Diederik P. Kingma and Jimmy Ba},
+ *   title   = {Adam: {A} Method for Stochastic Optimization},
+ *   journal = {CoRR},
+ *   year    = {2014},
+ *   url     = {http://arxiv.org/abs/1412.6980}
+ * }
+ * @endcode
+ */
+class AdamWUpdate
+{
+ public:
+  /**
+   * Construct the Adam update policy with the given parameters.
+   *
+   * @param epsilon The epsilon value used to initialise the squared gradient
+   *        parameter.
+   * @param beta1 The smoothing parameter.
+   * @param beta2 The second moment coefficient.
+   * @param weightDecay the rate at which the Update regularises the iterant
```

Comment:
The comment should start with a capital letter, and is missing a ``.``  at the end.